docs/fpc-key-dist.puml

```diff
@@ -1,97 +1,245 @@
+  Enclave1      -> TLCC_Enclave1    ++: check_committed_data(registration_message)
+  return true /'Enclave1 -> TLCC_Enclave1'/
+    note left
+        if false, abort
+    end note
+  Enclave1      -> TLCC_Enclave1    ++: can_endorse(ECC2_S_PK)
+    note right
+        TLCC checks the endorsement policy to determine
+        if the recipient enclave is an eligible endorser.
+    end note
+    note right
+        TLCC should check that the recipient enclave's MRENCLAVE
+        matches the chaincode definition's MRENCLAVE.
+        This could enable key distribution from an old cc version,
+        with a different MRENCLAVE.
+    end note
+  return true /'Enclave1 -> TLCC_Enclave1'/
+    note left
+        if false, abort
+    end note
+  Enclave1      -> Enclave1         :   check enclave chaincode definition against sealed data
+  Enclave1      -> Enclave1         :   check ECC2_S_PK, ECC_E_PK, attestation_report
+  Enclave1      -> Enclave1         :   m <- encrypt {SEK, CC_SK} with ECC2_E_PK
+    note left
+        enclave encrypts state encryption key and chaincode private key
+        - m denotes encrypted message
+    end note
+
+  Enclave1      -> Enclave1         :   SIG_ECC1 <- sign {ECC1_S_PK, ECC2_s_PK, m, CC_PK} with ECC1_S_SK
+    note left
+        enclave signs the key material it distributes together with the
+        sender and intended recipient identity
+    end note
+
+  return ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1 /'ECC1 -> Enclave1'/
+  return ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1 /'ECC_Manager1 -> ECC1'/
+  return ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1 /'Admin1 -> ECC_Manager1'/
+
+  note right Admin1
+    Key distribution is in progress. We need a means to transfer the message with
+    the encrypted keys to the recipient enclave. The next step uses the ledger
+    itself for this data transfer. This additionally allows to log the two phases
+    of the key distribution: (1) the sender enclave produced the encrypted data blob;
+    (2) the recipient enclave correctly received it. This can be useful when dealing
+    with enclaves of two different orgs.
+  end note
+
+  Admin1        -> ERCC1            ++: Start_KeyDist(ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1)
+  ERCC1         -> ERCC1            :   check ECC1_S_PK, ECC2_s_PK are registered and can endorse
+  ERCC1         -> ERCC1            :   check SIG_ECC1
+  ERCC1         -> ERCC1            :   put_state({"keydist_started", ECC1_S_PK, ECC2_s_PK, m, CC_PK, SIG_ECC1})
+  return /'Admin1 -> ERCC1'/
+end group
+
+group complete key distribution
+  Admin2        -> ECC_Manager2     ++: ConfirmCCKeyDist(CC_name, CC_ver, CH_id)
+  ECC_Manager2  -> ECC2             ++: ConfirmCCKeyDist(CC_name, CC_ver, CH_id)
+  ECC2          -> Enclave2         ++: ConfirmCCKeyDist(CC_name, CC_ver, CH_id)
+  Enclave2      -> ERCC2            ++: GetKeyDistStart(ECC2_S_PK)
```

Comment:
I guess you mean... letting ECCM2 do ECCM1's work. The thing is that they could belong to different Orgs, so Org1 would likely want to authorize this -- particularly if Org2 has lower standards/reputation. But this could also be solved by making Org1 delegating this step to Org2.
Anyway, in general however, I tried to keep the enclave management bound to the enclave's host.